Assets/Mirror/Editor/Weaver/Readers.cs

```diff
@@ -28,6 +29,38 @@ public static MethodReference GetReadFunc(TypeReference variable, int recursionC
             }
 
             TypeDefinition td = variable.Resolve();
+            #region function generator code
+
+            MethodDefinition newReaderFunc;
+
+            if (variable.IsArray)
+            {
+                newReaderFunc = GenerateArrayReadFunc(variable, recursionCount);
+            }
+            else if (td.IsEnum)
+            {
+                return GetReadFunc(td.GetEnumUnderlyingType(), recursionCount);
+            }
+            else if (variable.FullName.StartsWith("System.ArraySegment`1", StringComparison.Ordinal))
+            {
+                newReaderFunc = GenerateArraySegmentReadFunc(variable, recursionCount);
+            }
+            else
```

Comment:
I get that this fixes your case,  but this is not right,  this else catches will catch all unsupported types.

I'll fix this when I have a chance today.

Reply:
Didn't get that.What do you mean by  get all unsupported types?